prog/vnet/aws/vpc_nexus.rb

```diff
@@ -0,0 +1,193 @@
+# frozen_string_literal: true
+
+require "aws-sdk-ec2"
+class Prog::Vnet::Aws::VpcNexus < Prog::Base
+  subject_is :private_subnet
+
+  def before_run
+    when_destroy_set? do
+      if strand.label != "destroy"
+        register_deadline(nil, 10 * 60)
+        hop_destroy
+      end
+    end
+  end
+
+  label def start
+    PrivateSubnetAwsResource.create_with_id(private_subnet.id)
+    hop_create_vpc
+  end
+
+  label def create_vpc
+    vpc_response = client.describe_vpcs({filters: [{name: "tag:Name", values: [private_subnet.name]}]})
+
+    vpc_id = if vpc_response.vpcs.empty?
+      client.create_vpc({cidr_block: private_subnet.net4.to_s,
+        amazon_provided_ipv_6_cidr_block: true,
+        tag_specifications: Util.aws_tag_specifications("vpc", private_subnet.name)}).vpc.vpc_id
+    else
+      vpc_response.vpcs.first.vpc_id
+    end
+
+    private_subnet.private_subnet_aws_resource.update(vpc_id: vpc_id)
+    hop_wait_vpc_created
+  end
+
+  label def wait_vpc_created
+    vpc = client.describe_vpcs({filters: [{name: "vpc-id", values: [private_subnet.private_subnet_aws_resource.vpc_id]}]}).vpcs[0]
+
+    if vpc.state == "available"
```

Comment:
Up to you, but

```suggestion
    nap 1 unless vpc.state == "available"
```

and removing the extra indentation from the rest of the method may be simpler.